contracts/StakeHub.sol

```diff
@@ -1011,6 +1046,138 @@ contract StakeHub is SystemV2, Initializable, Protectable {
+        uint256 length = validatorNodeIDs[validator].length;
+        require(length > 0, "No NodeIDs registered for validator");
+
+        // Find the index of the target NodeID
+        uint256 indexToRemove = length; // default invalid index
+        for (uint256 i = 0; i < length; i++) {
+            if (validatorNodeIDs[validator][i] == targetNodeID) {
+                indexToRemove = i;
+                break;
+            }
+        }
+        require(indexToRemove < length, "NodeID not found for validator");
+
+        // Remove the NodeID using the swap-and-pop technique
+        if (indexToRemove != length - 1) {
+            validatorNodeIDs[validator][indexToRemove] = validatorNodeIDs[validator][length - 1];
+        }
+        validatorNodeIDs[validator].pop();
+        emit NodeIDRemoved(validator, targetNodeID);
+
+        // Clear the array if empty to save storage
+        if (validatorNodeIDs[validator].length == 0) {
+            delete validatorNodeIDs[validator];
+        }
+    }
+
+    /**
+     * @notice Replaces an existing NodeID with a new one for a validator.
+     * @param validator The address of the validator.
+     * @param oldNodeID The NodeID to be replaced.
+     * @param newNodeID The new NodeID to replace the old one with.
+     */
+    function replaceNodeID(
+        address validator,
+        bytes32 oldNodeID,
+        bytes32 newNodeID
+    ) external whenNotPaused notInBlackList validatorExist(validator) onlyAuthorized(validator) {
+        require(newNodeID != bytes32(0), "Invalid new NodeID");
+        require(oldNodeID != bytes32(0), "Invalid old NodeID");
+        require(newNodeID != oldNodeID, "New NodeID same as old NodeID");
+
+        uint256 length = validatorNodeIDs[validator].length;
+        require(length > 0, "No NodeIDs registered for validator");
+
+        // Check if newNodeID already exists
+        for (uint256 i = 0; i < length; i++) {
+            require(
+                validatorNodeIDs[validator][i] != newNodeID,
+                "New NodeID already exists"
+            );
+        }
+
+        // Find the index of the old NodeID
+        uint256 indexToReplace = length; // default invalid index
+        for (uint256 i = 0; i < length; i++) {
+            if (validatorNodeIDs[validator][i] == oldNodeID) {
+                indexToReplace = i;
+                break;
+            }
+        }
+        require(indexToReplace < length, "Old NodeID not found");
+
+        // Replace the NodeID
+        validatorNodeIDs[validator][indexToReplace] = newNodeID;
+        
+        emit NodeIDRemoved(validator, oldNodeID);
+        emit NodeIDAdded(validator, newNodeID);
+    }
+
+    /**
+     * @notice Returns all registered NodeIDs for a given validator.
+     * @param validator The address of the validator.
+     * @return An array of NodeIDs (bytes32[]).
+     */
+    function getNodeIDs(
```

Comment:
I think this method can be removed, because of the `listNodeIDsFor` can cover the secenario.